include/HTTPRequest.hpp

```diff
@@ -90,6 +90,24 @@ class HTTPRequest final
         const std::string& fileName = "",
         const std::unordered_set<std::string>& httpHeaders = DEFAULT_HEADERS);
 
+    /**
+     * @brief Performs an HTTP PATCH request.
+     *
+     * @param url URL to send the request.
+     * @param data Data to send.
+     * @param onSuccess Callback to be called when the request is successful.
+     * @param onError Callback to be called when an error occurs.
+     * @param fileName File name of output file.
+     * @param httpHeaders Headers to be added to the query.
+     */
```

Comment:
Note: Not for this PR because the other methods are the same, but you could use:
```suggestion
    /**
      * @copydoc IURLRequest::patch()
      */
```